src/evaluators/TryStatement.js

```diff
@@ -121,7 +121,7 @@ export default function(ast: BabelNodeTryStatement, strictCode: boolean, env: Le
         "composeNestedThrowEffectsWithHandler/1"
       );
     }
-    priorEffects.pop();
+    priorEffects.shift();
```

Comment:
If this fix is right then the `push` / `pop` pair below also needs changing.

I'm not sure the fix is right. "Pop" seems intended to match the "push" before it. It's strange to push first, and then shift later. Maybe the intention was to *remove the specific item we pushed*? In that case it would still be symmetrical but it would make sense if it broke when more than one item have been pushed since then.